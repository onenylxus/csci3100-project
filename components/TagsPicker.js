// Import
import React from 'react';
import { Text, View, Dimensions } from 'react-native';
import { Picker } from '@react-native-community/picker';
import Style from '../assets/style';
import TagsList from '../assets/json/tagsList.json';

// Export tag picker
export default function TagPicker({ callback, value }) {
  const windowWidth = Dimensions.get('window').width;

  function styleByDevice(widthOfDevice, component) {
    if (widthOfDevice < 800) {
      // Style of Small Screen
      switch (component) {
        case 'width':
          return '100%';

        case 'pickerWidth':
          return '100%';

        default:
          break;
      }
    }

    // Style of Large Screen
    switch (component) {
      case 'width':
        return 600;

      case 'pickerWidth':
        return 450;

      default:
        break;
    }
  }

  return (
    <View>
      {/* Tag */}
      <View
        style={{
          flexDirection: 'column',
          width: styleByDevice(windowWidth, 'width'),
        }}
      >
        <View style={Style.tagsTitle}>
          <Text style={{ alignSelf: 'center', marginVertical: 5 }}>Tag: </Text>
        </View>
        <View>
          <Picker
            style={{
              width: styleByDevice(windowWidth, 'pickerWidth'),
              alignSelf: 'center',
              marginVertical: 5,
            }}
            selectedValue={value}
            onValueChange={(itemValue) => callback(itemValue)}
          >
            {[
              <Picker.Item label="Choose a tag" value="" key="" />,
              ...Object.keys(TagsList).map((item) => (
                <Picker.Item label={TagsList[item]} value={item} key={item} />
              )),
            ]}
          </Picker>
        </View>
      </View>
    </View>
  );
}
