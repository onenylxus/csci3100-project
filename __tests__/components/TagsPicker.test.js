// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import TagsPicker from '../../components/TagsPicker';

// Run jest tests
describe('TagsPicker', () => {
  let element;

  beforeEach(() => {
    // Render
    element = render(<TagsPicker />);
  });

  afterEach(() => {
    element.unmount();
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});