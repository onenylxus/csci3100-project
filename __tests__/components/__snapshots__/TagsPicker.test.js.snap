// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsPicker matches snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "marginVertical": 5,
          "width": "20%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "marginVertical": 5,
          }
        }
      >
        Tag: 
      </Text>
    </View>
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "marginVertical": 5,
            "width": "100%",
          }
        }
      >
        <RNCPicker
          items={
            Array [
              Object {
                "label": "Choose a tag",
                "textColor": undefined,
                "value": "",
              },
              Object {
                "label": "Academics",
                "textColor": undefined,
                "value": "Academics",
              },
              Object {
                "label": "Relationships",
                "textColor": undefined,
                "value": "Relationships",
              },
              Object {
                "label": "News",
                "textColor": undefined,
                "value": "News",
              },
              Object {
                "label": "CU-Related",
                "textColor": undefined,
                "value": "CU-Related",
              },
              Object {
                "label": "Entertainment",
                "textColor": undefined,
                "value": "Entertainment",
              },
              Object {
                "label": "Sports",
                "textColor": undefined,
                "value": "Sports",
              },
              Object {
                "label": "Gaming",
                "textColor": undefined,
                "value": "Gaming",
              },
              Object {
                "label": "Fasion",
                "textColor": undefined,
                "value": "Fashion",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
